operator/src/lib.rs

```diff
@@ -393,10 +457,193 @@ impl NifiState {
 
         Ok((pod, config_maps))
     }
+
+    /// In order to enable / disable monitoring for NiFi, we have to make several REST calls.
+    /// There will be only one ReportingTask for the whole cluster. The task will be synced
+    /// for all nodes.
+    /// We always iterate over all the <node_name>:<http_port> pod combinations in order to
+    /// make sure that network problems etc. will not affect this. Usually the first pod
+    /// should be sufficient.
+    /// +-------------------------------------------------------------------------------+
+    /// |         "StackablePrometheusReportingTask" available?                         |
+    /// |            <no> |                          | <yes>                            |
+    /// |                 v                          v                                  |
+    /// |          metrics_port set                metrics_port set                     |
+    /// |       <no> |          | <yes>         <yes> |         | <no>                  |
+    /// |            v          v                     |         v                       |
+    /// | nothing to do       create                  |       status == running         |
+    /// |                                             |     <yes> |         | <no>      |
+    /// |                                             |           v         v           |
+    /// |                                             |    stop task      delete task   |
+    /// |                                             v                                 |
+    /// |                                  task_port == metrics_port                    |
+    /// |                                 <yes> |              | <no>                   |
+    /// |                                       v              v                        |
+    /// |                           status == stopped       status == running           |
+    /// |                          <yes> |              <yes> |         | <no>          |
+    /// |                                v                    v         v               |
+    /// |                             start task        stop task    delete task        |
+    /// +-------------------------------------------------------------------------------+
```

Comment:
Nice ascii art. I home cargo doc renders it properly. But does it have to be this complicated ? Why do you have to differentiate between "delete task" and "stop task" ? Can we not always delete the task (in the appropriate branches)?

Reply:
I dont think it renders properly, just had no other way to document the logic. 
For the delete and stop thing, you always have to stop the task before you edit or delete it as far i understood.
(In the NiFi UI you have to stop before editing and deleting as well)